k8s/migration/internal/controller/migration_controller.go

```diff
@@ -316,6 +341,105 @@ func newHostPathType(pathType string) *corev1.HostPathType {
 	return &hostPathType
 }
 
+func (r *MigrationReconciler) checkStatusSuccess(ctx context.Context,
+	migration *vjailbreakv1alpha1.Migration,
+	credsname string,
+	isvmware bool,
+	credsobj client.Object) (bool, error) {
+	err := r.Get(ctx, types.NamespacedName{Name: credsname, Namespace: migration.Namespace}, credsobj)
+	if err != nil {
+		return false, fmt.Errorf("failed to get VMwareCreds: %w", err)
+	}
+
+	if isvmware && credsobj.(*vjailbreakv1alpha1.VMwareCreds).Status.VMwareValidationStatus != success {
+		return false, fmt.Errorf("VMwareCreds '%s' CR is not validated", credsobj.(*vjailbreakv1alpha1.VMwareCreds).Name)
+	} else if !isvmware && credsobj.(*vjailbreakv1alpha1.OpenstackCreds).Status.OpenStackValidationStatus != success {
+		return false, fmt.Errorf("OpenstackCreds '%s' CR is not validated", credsobj.(*vjailbreakv1alpha1.OpenstackCreds).Name)
+	}
+	return true, nil
+}
+
+//nolint:dupl // Similar logic to networks reconciliation, excluding from linting to keep it readable
+func reconcileStorage(ctx context.Context,
+	migration *vjailbreakv1alpha1.Migration,
+	vmwcreds *vjailbreakv1alpha1.VMwareCreds,
+	openstackcreds *vjailbreakv1alpha1.OpenstackCreds,
+	vm string,
+	storages []vjailbreakv1alpha1.Storage) ([]string, error) {
+	vmds, err := GetVMwDatastore(ctx, vmwcreds, migration.Spec.Source.DataCenter, vm)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get datastores: %w", err)
+	}
+
+	openstackvolumetypes := []string{}
+	for _, vmdatastore := range vmds {
+		for _, storagemaptype := range storages {
+			if vmdatastore == storagemaptype.Source {
+				openstackvolumetypes = append(openstackvolumetypes, storagemaptype.Target)
+			}
+		}
+	}
+	if len(openstackvolumetypes) != len(vmds) {
+		return nil, fmt.Errorf("VMware Datastore(s) not found in StorageMapping")
+	}
+
+	err = VerifyStorage(ctx, openstackcreds, openstackvolumetypes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to verify datastores: %w", err)
+	}
+	return openstackvolumetypes, nil
+}
+
+func reconcileMapping(ctx context.Context,
+	migration *vjailbreakv1alpha1.Migration,
+	openstackcreds *vjailbreakv1alpha1.OpenstackCreds,
+	vmwcreds *vjailbreakv1alpha1.VMwareCreds,
+	vm string,
+	networks []vjailbreakv1alpha1.Network,
+	storages []vjailbreakv1alpha1.Storage) (openstacknws, openstackvolumetypes []string, err error) {
+	openstacknws, err = reconcileNetwork(ctx, migration, openstackcreds, vmwcreds, vm, networks)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to reconcile network: %w", err)
+	}
+
+	openstackvolumetypes, err = reconcileStorage(ctx, migration, vmwcreds, openstackcreds, vm, storages)
```

Comment:
It might be useful to update the `StorageMapping` status here, before we try to use it in a `Migration`. Thoughts?

Reply:
Fixed

Reply:
status:
      storageMappinValidationMessage: StorageMapping validated
      storageMappingValidationStatus: Succeeded